src/tosam.py

```diff
@@ -1,18 +1,21 @@
-import argparse
-
-
-def main():
-    argparser = argparse.ArgumentParser(description="To Simple-SAM converter")
-    argparser.add_argument(
-        "mas",
-        type=argparse.FileType('r')
-    )
-    args = argparser.parse_args()
-    print(args)
-    for line in args.mas:
-        chrom, read_name, read_str, pos = line.split('\t')
-        # Output as Simple-SAM
-
-
-if __name__ == '__main__':
-    main()
+import argparse
+import sys
+
+parser = argparse.ArgumentParser(prog = "tosam")
+parser.add_argument("-path", nargs = 1, type = str)

```

Comment:
Who told you that the file name argument should have a flag such as `-path`? I'm pretty sure I didn't, and the test I wrote assumes that it can just call the tool with a positional arguments. If you are going to mess with the arguments, ensure that you satisfy the requirements the exercise has for your tool. Otherwise, you will not pass the automated tests, and if that is for a project, you will have to redo your code if you want to qualify for the exam. I don't consider changing options such a small matter in this regard. There is a specification, and you have to follow it.

You probably shouldn't use strings for file arguments either. The `argparse` module is good at handling files, it handles when a user use `-` to mean `stdin`/`stdout`, and it handles missing arguments with defaults. There are very few occasions where I can imagine it being better to implement that functionality yourself. This isn't one of them. This is your choice, of course, and you are welcome to do it this way, but I suggest you consider the "right" way as well.